src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -815,6 +818,10 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
                     pattern = rList[0];
                     if (rList.Count > 1)
                     {
+                        if (!(rList[1] is string))
+                        {
+                            useMatchEvaluator = LanguagePrimitives.TryConvertTo(rList[1], out matchEvaluator);
+                        }
                         replacement = PSObject.ToStringParser(context, rList[1]);
```

Comment:
Should this line be run if we use a matchEvaluator?

Reply:
[`TryConvertEmpty()` will do a simple cast and return early](https://github.com/PowerShell/PowerShell/blob/a9c6292903d86a7bf3b3bfb59110d03d1234c525/src/System.Management.Automation/engine/LanguagePrimitives.cs#L1681) if the input object and out param are the same type, no need to complicate the `ReplaceOperator()` implementation for this (edge-)case